### PR TITLE
feat(wasm-gen): allow `gr_send_commit*` to receive existing address

### LINKED ISSUE #3271: Allow `gr_send_commit*` to receive existing address in `wasm-gen`
### Problem to Solve

Currently it receives random data for destination. Should be the same as it's done for `gr_send` and etc.

### PR DESCRIPTION
**Pull request by StackOverflowExcept1on**
Resolves #3271

@gear-tech/dev

### CODE REVIEW

File: utils/wasm-gen/src/generator/syscalls/invocator.rs [hunk 1]
```diff
@@ -167,6 +167,25 @@ impl ParamSetter {
 
 pub type SysCallInvokeInstructions = Vec<Instruction>;
 
+pub enum SysCallKind {
+    WithDestination(usize),
+    Common,
+}
+
+impl From<InvocableSysCall> for SysCallKind {
+    fn from(invocable: InvocableSysCall) -> Self {
```
Comment (mertwole) [inline utils/wasm-gen/src/generator/syscalls/invocator.rs:176]: I think it'll be better to implement method on the `InvocableSysCall`, smth like:
```rust
fn has_destination_param(&self) -> Option<usize> { ... }
```

Comment (mertwole) [reply utils/wasm-gen/src/generator/syscalls/invocator.rs:176]: As `SysCallKind` looks a way more general

**Review by mertwole — CHANGES_REQUESTED** [review]

File: utils/wasm-gen/src/generator/syscalls/invocator.rs (hunk 1 above)
Comment (StackOverflowExcept1on) [reply utils/wasm-gen/src/generator/syscalls/invocator.rs]: done in c94c76d501cd5b3e49bf47fdf36b8943c41bfc96

**Review by breathx — APPROVED** [review]

**Comment by breathx** [comment]
@SabaunT

**Review by SabaunT — APPROVED** [review]
better than mutating syscall signature struct

**Pull request merged**